tests/helper/helper_oc.go

```diff
@@ -766,3 +767,71 @@ func (oc OcRunner) CreateSecret(secretName, secretPass, project string) {
 func (oc OcRunner) GetSecrets(project string) string {
 	return GetSecrets(oc.path, project)
 }
+
+// AddSecret adds pull-secret to the namespace, for e2e-test
+func (oc OcRunner) AddSecret(comvar CommonVar) {
+
+	clusterType := os.Getenv("CLUSTER_TYPE")
+	if clusterType == "PSI" || clusterType == "IBM" {
+
+		token := oc.doAsAdmin(clusterType)
+
+		yaml := Cmd(oc.path, "get", "secret", "pull-secret", "-n", "openshift-config", "-o", "yaml").ShouldPass().Out()
+		newYaml := strings.Replace(yaml, "openshift-config", comvar.Project, -1)
+		filename := fmt.Sprint(RandString(4), ".yaml")
+		f, err := os.Create(filename)
+		if err != nil {
+			fmt.Println(err)
+		}
+		_, err = f.WriteString(newYaml)
+		if err != nil {
+			fmt.Println(err)
+			f.Close()
+		}
+		err = f.Close()
```

Comment:
You could use ioutil.WriteFile instead (https://gobyexample.com/writing-files)

Reply:
Is this what you are suggesting? 
```
filename := fmt.Sprint(RandString(4), ".yaml")
d1 := []byte(newYaml)
err := ioutil.WriteFile(filename, d1, 0644)
if err != nil {
     panic(err)
}		
Cmd(oc.path, "apply", "-f", filename).ShouldPass()
```

Reply:
Yes, right

Reply:
done